Move remaining uses of serviceName away from urlParams

smith opened the pull request:
There were a few of these that were either missed or lost in merge conflict resolution.

I went through everything that's used as a path parameter and made sure it wasn't being used anywhere with `urlParams`.

Previously none of the charts were working, now they all are.

smith committed:
Move remaining uses of serviceName away from urlParams

There were a few of these that were either missed or lost in merge conflict resolution.

I went through everything that's used as a path parameter and made sure it wasn't being used anywhere with `urlParams`.

Previously none of the charts were working, now they all are.

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/)
* Commit: 79e47dda826a98af64e8b04dc694dce2d618d08f

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/79e47dda826a98af64e8b04dc694dce2d618d08f) | diff | [baseline](https://github.com/elastic/kibana/commit/eacd602612fc80f8aa691988fcf7bcdd34607cef) |
| --- | --- | --- | --- |
| apm | 4.9MB | -1.3KB | 4.9MB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/","number":"73868","commit":"79e47dda826a98af64e8b04dc694dce2d618d08f"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/public/components/app/RumDashboard/CoreVitals/index.tsx:
```diff
@@ -20,10 +20,11 @@ const CoreVitalsThresholds = {
 export function CoreVitals() {
   const { urlParams, uiFilters } = useUrlParams();
```
Should we rename this to make the distinction between path params (that we get from react-router) and query params clear?

```suggestion
  const { queryParams, uiFilters } = useQueryParams();
```

smith replied:
I'd be fine with doing that but we would have to change it in a hell of a lot of places. I'd prefer doing that change in the next round of changes and leaving it for now.

smith commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into nls/missing-params

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/missing-params

smith committed:
Use useLocation

smith commented on x-pack/plugins/apm/public/components/shared/DatePicker/index.tsx:
```diff
@@ -7,10 +7,9 @@
 import { isEmpty, isEqual, pickBy } from 'lodash';
 import React from 'react';
-import { useHistory } from 'react-router-dom';
+import { useHistory, useLocation } from 'react-router-dom';
```
Replacing our `useLocation` here with the one from `react-router-dom`. This is also fixed in #76343 but CSM is not using our old location provider.

shahzad31 reviewed (approved):
LGTM CSM changes

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/)
* Commit: 6cb620d4212a7a3d3877827ec20870b1ff021f59

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/6cb620d4212a7a3d3877827ec20870b1ff021f59) | diff | [baseline](https://github.com/elastic/kibana/commit/0c3a8c5f4ef4735ba5a784c6e916b3289834682b) |
| --- | --- | --- | --- |
| apm | 4.9MB | -1.4KB | 4.9MB |

### History
* :green_heart: [Build #73868](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/) succeeded 79e47dda826a98af64e8b04dc694dce2d618d08f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/","number":"73868","commit":"79e47dda826a98af64e8b04dc694dce2d618d08f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/","number":"74184","commit":"6cb620d4212a7a3d3877827ec20870b1ff021f59"}]}
PIPELINE-->

smith commented:
retest

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74269/)
* Commit: 6cb620d4212a7a3d3877827ec20870b1ff021f59

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/6cb620d4212a7a3d3877827ec20870b1ff021f59) | diff | [baseline](https://github.com/elastic/kibana/commit/0c3a8c5f4ef4735ba5a784c6e916b3289834682b) |
| --- | --- | --- | --- |
| apm | 4.9MB | -1.4KB | 4.9MB |

### History
* :green_heart: [Build #74184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/) succeeded 6cb620d4212a7a3d3877827ec20870b1ff021f59
* :green_heart: [Build #73868](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/) succeeded 79e47dda826a98af64e8b04dc694dce2d618d08f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/","number":"73868","commit":"79e47dda826a98af64e8b04dc694dce2d618d08f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/","number":"74184","commit":"6cb620d4212a7a3d3877827ec20870b1ff021f59"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74269/","number":"74269","commit":"6cb620d4212a7a3d3877827ec20870b1ff021f59"}]}
PIPELINE-->

smith committed:
Fix APM e2e tests

Looks like #67791 introduced a find/replace change that broke APM's e2e tests. This reverts that change.

smith committed:
Remove remaining hashes

smith committed:
Merge branch 'nls/feature-fix' into nls/missing-params

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74329/)
* Commit: d656820518be4d6d4008cf1d405266cc94192bf3

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/d656820518be4d6d4008cf1d405266cc94192bf3) | diff | [baseline](https://github.com/elastic/kibana/commit/525660413e6f802957e1c97c7484c5e1b0603c8b) |
| --- | --- | --- | --- |
| apm | 4.9MB | -1.4KB | 5.0MB |

### History
* :green_heart: [Build #74269](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74269/) succeeded 6cb620d4212a7a3d3877827ec20870b1ff021f59
* :green_heart: [Build #74184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/) succeeded 6cb620d4212a7a3d3877827ec20870b1ff021f59
* :green_heart: [Build #73868](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/) succeeded 79e47dda826a98af64e8b04dc694dce2d618d08f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73868/","number":"73868","commit":"79e47dda826a98af64e8b04dc694dce2d618d08f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74184/","number":"74184","commit":"6cb620d4212a7a3d3877827ec20870b1ff021f59"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74269/","number":"74269","commit":"6cb620d4212a7a3d3877827ec20870b1ff021f59"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74329/","number":"74329","commit":"d656820518be4d6d4008cf1d405266cc94192bf3"}]}
PIPELINE-->
